### PR TITLE
Add API to manage repo tranfers

### PR DESCRIPTION
**Pull request by qwerty287**
Get/accept/reject transfers via API.

New endpoints:
* `POST api/v1/repos/{owner}/{repo}/transfer/accept` accept transfer
* `POST api/v1/repos/{owner}/{repo}/transfer/reject` reject transfer

Updated endpoints:
* `GET api/v1/repos/{owner}/{repo}` added `RepoTransfer` object that contains information about a transfer (`null` if no transfer is available)

### CODE REVIEW

File: modules/structs/repo.go
Comment (lunny) [inline modules/structs/repo.go:96]: Why we need this?

Comment (qwerty287) [reply modules/structs/repo.go:96]: To have a possibility for clients to check if the repo is currently waiting for an approval to transfer it

File: modules/convert/repository.go
Comment (6543) [inline modules/convert/repository.go]:
```suggestion
		if err != nil && !models.IsErrNoPendingTransfer(err) {
```

File: routers/api/v1/repo/transfer.go
Comment (6543) [inline routers/api/v1/repo/transfer.go:209]: if IsErrNoPendingTransfer(err) -> 404

File: routers/api/v1/repo/transfer.go
Comment (6543) [inline routers/api/v1/repo/transfer.go]:
```suggestion
	if accept {
		return repo_service.TransferOwnership(repoTransfer.Doer, repoTransfer.Recipient, ctx.Repo.Repository, repoTransfer.Teams)
	}

	return models.CancelRepositoryTransfer(ctx.Repo.Repository)
```

**Review by 6543 — APPROVED** [review]

**Comment by qwerty287** [comment]
@6543 thanks for adding the 404 handler

**Comment by 6543** [comment]
@qwerty287 you could add this case to the integration tests ... but you already have my lgtm ;)

**Comment by qwerty287** [comment]
@6543 see https://github.com/go-gitea/gitea/pull/17963/commits/f695dc65c8faa8419236016be293f11379a0e51d

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
Linter throw an error

**Comment by qwerty287** [comment]
Could we move this to 1.16 since it's ready and approved (once)?

File: modules/convert/repository.go
Comment (wxiaoguang) [inline modules/convert/repository.go]: Here and above, I think we can log these unexpected errors as WARN or ERROR, it makes debug easier.

**Review by wxiaoguang — APPROVED** [review]

File: modules/convert/repository.go
Comment (qwerty287) [reply modules/convert/repository.go]: https://github.com/go-gitea/gitea/pull/17963/commits/70841452d48b0b3eebc6aca8647e7e7e8baa74a2

**Pull request merged**
